src/types/widget.ts

```diff
@@ -6,18 +6,55 @@ import {
   SearchParameters,
 } from './instantsearch';
 
-interface WidgetArgs {
-  helper: Helper;
+interface InitOptions {
+  instantSearchInstance: InstantSearch;
   state?: HelperState;
-  results?: SearchResults;
+  helper?: Helper;
+  templatesConfig?: object;
+  createURL?(state: HelperState): string;
```

Comment:
Almost all the attributes are required.

https://github.com/algolia/instantsearch.js/blob/d553502f13e7eb9b6d423f85983c19e5732cc7a5/src/lib/InstantSearch.js#L414-L421